pkgs/development/ocaml-modules/capnp/default.nix

```diff
@@ -0,0 +1,50 @@
+{ pkgs
+, lib
+, buildDunePackage
+, fetchFromGitHub
+, capnproto
+, ocplib-endian
+, res
+, result
+, stdint
+, stdio
+}:
+
+buildDunePackage rec {
+  pname = "capnp";
+  version = "3.6.0";
+
+  minimalOCamlVersion = "4.08";
+
+  src = fetchFromGitHub {
+    owner = "capnproto";
+    repo = "capnp-ocaml";
+    rev = "v${version}";
+    hash = "sha256-G4B1llsHnGcuGIarDB248QMaRBvS47IEQB5B93wY7nA=";
+  };
+
+  nativeBuildInputs = [
+    capnproto
+  ];
+
+  buildInputs = [
+    result
```

Comment:
The `result` library belongs to `propagatedBuildInputs`.

Reply:
Sounds good -- fixed!

I have a hard time figuring out which build inputs need to be propagated.

How did you figure this out?

Reply:
Excellent! Thanks for the tip!